keps/sig-apps/1847-autoremove-statefulset-pvcs/README.md

```diff
@@ -394,10 +399,17 @@ This features adds a new field to the StatefulSet. The default value for the new
 maintains the existing behavior of StatefulSets.
 
 On a downgrade, the `PersistentVolumeClaimRetentionPolicy` field will be hidden on
-any StatefulSets. The behavior in this case will be identical to mutating they
+any StatefulSets. The behavior in this case will be identical to mutating the
 policy field to `Retain`, as described above, including the edge cases
 introduced if this is done during a scale-down or StatefulSet deletion.
 
+The initial beta version did not set the controller flag in the owner
+reference. If a PVC has an owner reference to the expected StatefulSet or Pod
+object, and the controller flag is not set, it will be set, if there are no
+other controllers on the object. If there is another controller, the existing
+non-controller StatefulSet or Pod reference will be removed. That is, we will
```

Comment:
Could you please elaborate on this?  That is, what is the meaning behind foreign controller references and our contract to the PVC.

Reply:
The current beta implementation does not set the controller flag. So we need to deal with the case that the owner ref has been (incorrectly) set by the current version of the controller.

I'd be happy to have other opinions on how to handle this --- the use of controller references does not seem very well specified.

Reply:
Ah, I think I got a bit confused about which controller is which. Can you at least mention the StatefulSet controller by its full name when talking about it? And probably 3rd party PVC controller when talking about the other one?

I suppose the paragraph captures the meaning okay. The key takeaways are:

#### If we are the controller (StatefulSet)
- fully manage the PVC lifecycle
- update our old owner references with the `.controller=false` to `controller=true`
- remove itself as the owner and controller when the retain policy is specified in the StatefulSet

#### If someone else is the controller
- let them fully manage the PVC lifecycle. 
- remove our old owner references
- the PVC will stay when the delete policy is specified  in the StatefulSet

I think there is a possibility for us to be an co-owner as well (delete policy), when somebody else is the owning controller. But I am not sure if there is any real advantage in that.

Reply:
> Ah, I think I got a bit confused about which controller is which. Can you at least mention the StatefulSet controller by its full name when talking about it? And probably 3rd party PVC controller when talking about the other one?

Good point, I didn't notice the ambiguity.

I like your bulleted approach, I'll add that in.

I think there are a lot of ways for us to co-own, but given the general lack of clarity on what controllers might do, I think it's dangerous. We don't know if an owning controller might have expectations on the lifetime of a PVC.

Reply:
It was just an idea. I agree, it is safer not to do that.

Reply:
this was done in the design details section as well.